connection-api/src/main/java/org/terracotta/connection/entity/EntityRef.java

```diff
@@ -26,27 +26,48 @@
 
 
 /**
- * The EntityRef is a reference to where an entity could potentially exist.  That is, it is created to refer to a specific
- * entity type and name, although there may not yet be an entity with that type and name.
- * 
- * The EntityRef is used to manage the server-side entity's life cycle in that it is how the entity is either created or
- * destroyed.
- * 
- * It is also used as the path through which an actual instance can be fetched for use on the client.
- * 
+ * The EntityRef is a reference to where an entity could potentially exist.
+ * <P>
+ *   That is, it is created to refer to a specific entity type and name, although there may not yet be an entity with
+ *   that type and name.
+ * </P>
+ * <P>
+ *   The EntityRef is used to manage the server-side entity's life cycle in that it is how the entity is either created
+ *   or destroyed.
+ * </P>
+ * <P>
+ *   It is also used as the path through which an actual instance can be fetched for use on the client.
+ * </P>
+ *
  * @param <T> The entity type underlying this reference.
  * @param <C> The configuration type to use when creating this entity.
  */
 public interface EntityRef<T extends Entity, C> {
   /**
-   * Create the entity with the given configuration.
+   * Creates the entity with the given configuration.
+   * <P>
+   *   If the entity already exists, the {@link EntityAlreadyExistsException} is  thrown immediately, whether or not
+   *   the entity is fetched on any client.
+   * </P>
+   * <P>
+   *   If a {@code create} is invoked while another {@code create} is happening, it will block until either:
+   *   <UL>
+   *     <LI>the other create completes and then throw an {@code EntityAlreadyExistsException}</LI>
+   *     <LI>the other create fails and then it will (try to) perform the create as well</LI>
+   *   </UL>
+   *   The contract provides guarantee that when multiple clients race to {@code create}, and in the absence of any
+   *   {@link #destroy()} or server failure, when the {@code create} method completes (returns or throws), a subsequent
+   *   {@link #fetchEntity()} will not throw an {@link EntityNotFoundException}.
+   * </P>
    *
    * @param configuration configuration to be applied to the entity
-   * @throws EntityNotProvidedException The service providing T doesn't exist on either the client or the server
```

Comment:
It can still throw if there is no provider on the server, though.  This is effectively an incomplete server installation.